pkg/util/cloudfoundry/cccache.go

```diff
@@ -0,0 +1,166 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// +build clusterchecks
+
+package cloudfoundry
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/cloudfoundry-community/go-cfclient"
+)
+
+// CCCacheI is an interface for a structure that caches and automatically refreshes data from Cloud Foundry API
+// it's useful mostly to be able to mock CCCache during unit tests
+type CCCacheI interface {
+	LastUpdated() time.Time
+	GetPollAttempts() int
+	GetPollSuccesses() int
+	GetApp(string) (*cfclient.V3App, error)
+}
+
+// CCCache is a simple structure that caches and automatically refreshes data from Cloud Foundry API
+type CCCache struct {
+	sync.RWMutex
+	cancelContext context.Context
+	configured    bool
+	ccAPIClient   CCClientI
+	pollInterval  time.Duration
+	pollAttempts  int
+	pollSuccesses int
+	lastUpdated   time.Time
+	appsByGUID    map[string]*cfclient.V3App
+}
+
+type CCClientI interface {
+	ListV3AppsByQuery(url.Values) ([]cfclient.V3App, error)
+}
+
+var (
+	globalCCCache     = &CCCache{}
+	globalCCCacheLock sync.Mutex
+)
+
+// ConfigureGlobalCCCache configures the global instance of CCCache from provided config
+func ConfigureGlobalCCCache(ctx context.Context, ccURL, ccClientID, ccClientSecret string, skipSSLValidation bool, pollInterval time.Duration, testing CCClientI) (*CCCache, error) {
+	globalCCCacheLock.Lock()
```

Comment:
Could you use the ccache lock directly instead of a different lock?

Reply:
I guess so ?